src/main/java/core/basesyntax/shopimpl/service/IOdataFileService.java

```diff
@@ -0,0 +1,75 @@
+package core.basesyntax.shopimpl.service;
+
+import core.basesyntax.model.shopstrategy.ShopTransactionsTypes;
+import core.basesyntax.shopimpl.entity.DataRecord;
+import core.basesyntax.shopimpl.entity.Fruit;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardOpenOption;
+import java.util.ArrayList;
+import java.util.List;
+
+public class IOdataFileService {
+    private static final String FILE_DELIMITER = ",";
+    private static final int FILE_INDEX_OF_ACTION = 0;
+    private static final int FILE_INDEX_OF_ITEM = 1;
+    private static final int FILE_INDEX_OF_AMOUNT = 2;
+    private final String dataFilePath;
```

Comment:
you have only one field with dataFilePath, but you should write and read from different files. I guess you changing this value somewhere; I would suggest you to pass files to writeDataFile and readDataFile along with data

Reply:
I changed the implementation of the class so it accepts the file path every time